session.go

```diff
@@ -718,6 +818,43 @@ func (s *session) handleUnpackedPacket(packet *unpackedPacket, rcvTime time.Time
 
 	r := bytes.NewReader(packet.data)
 	var isAckEliciting bool
+
+	if s.mitigationOn(packet.encryptionLevel) {
+		var parsedFirstFrame bool
+		var delayedErrorFrame wire.Frame
+		var delayedError *qerr.QuicError
+
+		validatingReader := bytes.NewReader(packet.data)
+		for {
+			frame, err := s.frameParser.ParseNext(validatingReader, packet.encryptionLevel)
+			if err != nil {
+				return err
+			}
+			if frame == nil {
+				if !parsedFirstFrame && !updateDestConnID {
+					return fmt.Errorf("packet contains only padding frames")
```

Comment:
Why is this only true if `!updateDestConnID`?  Are Retry packets permitted to contain padding?

Reply:
We could - this would mean that "firstPacketReceived" would be set to true, so an attacker who could inject this kind of packet could cause the connection to fail if the server wanted to send a Retry packet. But an attacker could also send a legit Initial packet and get the same result. It's a question of whether its significantly easier for an attacker to construct an all-padding Initial vs a real Initial, and in practice it's probably not that different

Reply:
It might be nice to have a timer that starts when we think we have received the first packet and doesn't block others until the timer is up, but that's probably beyond the scope of this PR